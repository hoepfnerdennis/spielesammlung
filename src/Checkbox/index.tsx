import React from 'react';
import styles from './styles.module.scss';

const Checkbox: React.SFC<{
  label: string;
  onChange: (value: string | undefined) => void;
  checked: boolean;
}> = ({ label, onChange, checked }): JSX.Element => {
  const handleClick = (): void => {
    onChange(!checked ? 'true' : undefined);
  };

  return (
    <button
      type="button"
      className={checked ? styles.checked : styles.unchecked}
      onClick={handleClick}>
      {label}
    </button>
  );
};

export default Checkbox;
