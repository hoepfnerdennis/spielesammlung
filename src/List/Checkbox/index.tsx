import React from 'react';
import styles from './styles.module.scss';

const Checkbox: React.SFC<{
  label: string;
  onChange: (value: boolean) => void;
  checked: boolean;
}> = ({ label, onChange, checked }): JSX.Element => {
  const handleClick = (): void => {
    onChange(!checked);
  };
  return (
    <button
      type="button"
      className={checked ? styles.checked : styles.unchecked}
      onClick={handleClick}>
      {label}
    </button>
  );
};

export default Checkbox;
