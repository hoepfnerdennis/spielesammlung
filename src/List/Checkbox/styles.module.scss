%base {
    height: 2rem;
    font-size: 1rem;
    border: 2px solid darkcyan;
    border-radius: 5px;
  }
  
.checked {
    @extend %base;
    background-color: darkcyan;
    color: white;
  }
  
  .unchecked {
    @extend %base;
    background-color: white;
    color: darkcyan;
}