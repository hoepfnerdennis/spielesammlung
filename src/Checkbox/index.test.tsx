import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import Checkbox from './index';

describe('Checkbox', () => {
  it.each([
    ['checked', true],
    ['unchecked', false],
  ])('should render %s', (_, checked) => {
    const onChange = jest.fn();
    const { container } = render(<Checkbox label="label" onChange={onChange} checked={checked} />);
    expect(container).toMatchSnapshot();
  });

  it('should handle click', () => {
    const onChange = jest.fn();
    const { getByText } = render(<Checkbox label="label" onChange={onChange} checked={false} />);
    fireEvent.click(getByText('label'));
    expect(onChange).toHaveBeenLastCalledWith(true);
  });
});
