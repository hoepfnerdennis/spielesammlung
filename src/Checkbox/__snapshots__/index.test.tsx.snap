// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render checked 1`] = `
<div>
  <button
    class="checked"
    type="button"
  >
    label
  </button>
</div>
`;

exports[`Checkbox should render unchecked 1`] = `
<div>
  <button
    class="unchecked"
    type="button"
  >
    label
  </button>
</div>
`;
